Core/src/IfcGeometryConverter/CurveConverter.h

```diff
@@ -2895,7 +2901,378 @@ namespace OpenInfraPlatform
+					{
+						double t = integrateParameter(polynomialConstantY, polynomialConstantZ, parameter);
+						y = calculatePolynomialCurve(polynomialConstantY, t);
+						z = calculatePolynomialCurve(polynomialConstantZ, t);
+						return carve::geom::VECTOR(0., y, z);
+					}
+
+					else if (polynomialConstantCntX > 0 && polynomialConstantCntY > 0 && polynomialConstantCntZ > 0)
+					{
+						double t = integrateParameter(polynomialConstantX, polynomialConstantY, polynomialConstantZ, parameter);
+						x = calculatePolynomialCurve(polynomialConstantX, t);
+						y = calculatePolynomialCurve(polynomialConstantY, t);
+						z = calculatePolynomialCurve(polynomialConstantZ, t);
+						return carve::geom::VECTOR(x, y, z);
+					}
+				}
+
+				/*! \brief Calculates a polinomial curve
+				* \param[in] coefficient			A vector with coeffitients corresponding ordinate from \c IfcPolynomialCurve.
+				* \param[in] parameter				A length during the curve from \c IfcCurveSegment.
+				* \return							The location of the ordinate point.
+				* \note
+				*/
+				typename IfcEntityTypesT::IfcReal calculatePolynomialCurve(std::vector<double>& coefficient, const double& parameter) const noexcept(false)
+				{
+					int n = std::size(coefficient);
+					typename IfcEntityTypesT::IfcReal sum = coefficient[0];
+					double factor = 1;
+
+					for (int i = 1; i < n; i++)
+					{
+						factor *= parameter;
+						sum += coefficient[i] * factor;
+					}
+					return sum;
+				}
+
+				/*! \brief Calculates derivative of the polynomial curve
+				* \param[in] polynomialConstant	    A vector with coeffitients corresponding ordinate from \c IfcPolynomialCurve.
+				* \param[in] b				        Parameter value from intagrateParameter
+				* \return							Derivative value.
+				* \note
+				*/
+				double calculatePolynomialDerivative(std::vector<double>& polynomialConstant, double b)  const noexcept(false)
+				{
+					int polynomialConstantCnt = std::size(polynomialConstant);
+					double value = polynomialConstant[1];
+					int factor = 2;
+					// derivative of the polynomial
+					for (int i = 2; i < polynomialConstantCnt; i++)
+					{
+						value += polynomialConstant[i] * factor * b;
+						b *= b;
+						factor += 1;
+					}
+					return value;
+				}
+				/*! \brief Calculates parameter for parametric polynomial curve
+				* \param[in] polynomialConstantX	A vector with coeffitients axis X from \c IfcPolynomialCurve.
+				* \param[in] polynomialConstantY	A vector with coeffitients axis Y from \c IfcPolynomialCurve.
+				* \param[in] length				    Length of the curve cegment
+				* \return							Parameter value.
+				* \note
+				*/
+				double integrateParameter(std::vector<double>& polynomialConstantX, std::vector<double>& polynomialConstantY, const double& length) const noexcept(false)
+				{
+					double a = 0.;
+					double b = 0.;
+					int n = 1;// small numbers return better values
+					double value = 0.;
+
+					if (length == 0) { return b; }

```

Comment:
in the following lines the curly braces are not used when returning b 
Try to be consistent and use them everywhere or leave them out completely